Tech debt webpack

wwelling opened the pull request:
Blocks issues

- https://github.com/TAMULib/DirectoryUI/issues/259
- https://github.com/TAMULib/CAP/issues/173
- https://github.com/TAMULib/SAGE/issues/422
- https://github.com/TAMULib/GIFMButtonUI/issues/48
- https://github.com/TAMULib/MagpieUI/issues/202
- https://github.com/TAMULib/SubjectLibrarianUI/issues/74
- https://github.com/TAMULib/LibraryServiceStatusSystemUI/issues/212
- https://github.com/TAMULib/ProjectManagementUI/issues/118
- https://github.com/TAMULib/MyLibraryUI/issues/318

jeremythuff committed:
WIP building dist

jeremythuff committed:
WIP concat js

jeremythuff committed:
Added clean cmd and taking config from app

jeremythuff committed:
Implemented williams sollution for concat

Commit:
update build with bundle dependencies

Commit:
only log warning and error

Commit:
update webserver to use dist directory

Commit:
fix build with plugin

Commit:
include view in dist

Commit:
fix build workflow

Commit:
add weaver resources

Commit:
move all dev dependencies and initial sass loader

Commit:
bundle dependencies and use wvr resource path

Commit:
bundle dependencies

Commit:
add resolve url loader

Commit:
add bootstrap sass to bundled dependencies

jeremythuff committed:
WIP no bundled deps

Commit:
alpha 20

jeremythuff committed:
WIP extracting css

jeremythuff committed:
Merge branch 'tech-debt-webpack' of github.com:TAMULib/Weaver-UI-Core into tech-debt-webpack

Commit:
afford config copy patterns

Commit:
revert bootstrap

Commit:
production undefined devtool

Commit:
add bootstrap fonts

Commit:
make no opinions on what to copy

Commit:
minify for production

Commit:
2.2.2-rc.3

Commit:
2.2.2-rc.4 as next

Commit:
use env coveralls repo secret

Commit:
cleanup formatting

Commit:
fix stop server

Commit:
more cleanup

Commit:
2.2.2-rc.5

Commit:
2.2.2-rc.6 as next

Commit:
update gitignore

Commit:
use repository secret

kaladay commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@wvr/core",
-  "version": "2.2.2-rc.2",
+  "version": "2.2.2-rc.6",
```
Bump this to `2.2.2-rc.8`? (or to whatever the latest version is by the time this PR is ready to go.)

wwelling replied:
Thanks. Will check.

wwelling replied:
Required version: https://github.com/TAMULib/MyLibraryUI/pull/319/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

wwelling replied:
Will be 2.2.8-rc.1 for development. Not yet published. Please checkout weaver-webpack-upgrade.

Commit:
add relevant comments to weaver default webpack config

Commit:
Merge branch '2.x' into tech-debt-webpack

kaladay committed:
Add support for running a local NPM Registry via Verdaccio. (#189)

* Add support for running a local NPM Registry via Verdaccio.

A single new npm run command is added "start:npm-local".
This will start an Verdaccio Registry and handle only the `@wvr` packages.
All other packages are proxied.
Custom settings are added to ensure that this does not cache proxied packages (the packages from the upstream NPM Registry).

This is designed to operate anonymously.
A connection to the local Verdaccio NPM Registry should not need an account.
The more recent versions of NPM have made this more difficult by removing anonymous support.
This is worked around by adding a .npmrc that injects a fake token specifically for the localhost registry.o

This utilizes a verdaccio-memory module to have verdaccio store the registry in memory, keeping the local filesystem clean.
This can be disabled by uncommenting the the appropriate section.
The rest of the registry is already configured to work if the in-memory support is disabled.
If uncommented, then be sure to handle cleaning up the .verdaccio directory yourself.

I wanted to have the following in the package.json:
```
  "publishConfig": {
    "access": "public",
    "registry": "http://localhost:4873/"
  },
```
Which would also necessitate the following in `.npmrc`:
```
registry=http://localhost:4873/
```

This cannot happen because the NPM developers have yet to realize the usefullness of a command line argument like `--registry`:
```
npm publish --registry https://registry.npmjs.org
```

The lack of the command like argument necessitates manually setting the localhost registry:
```
npm registry set http://localhost:4873/
```

This increases the possibility of mistakes, so I suggest always performing a dry run as a practice before actual publishing:
```
npm publish --dry-run
```

Once done with the local Verdaccio NPM Registry, you can reset the registry like this:
```
npm set registry https://registry.npmjs.org/
```

There are also additional configurations that reduce the needed changes for custom situations.
1) The `allow_offline` is enabled because localhost interaction should not require an active internet connection.
2) The `security.*` settings are provided in case one wants to set `web.login` to `true`.

* working on windows

* Using ':pre' and ':post' is not cross-system, instead prepend with 'pre' and 'post'.

Co-authored-by: William Welling <8352733+wwelling@users.noreply.github.com>

Commit:
Tech debt webpack replace forever and forever-monitor dev dependencies with webpack-dev-server. (#190)

* use webpack dev server

* more cleanup
